### PR TITLE
Cleanup network audit whitelist 

### LINKED ISSUE #7789: Remove old entries from the network auditor whitelist
The following entries are probably no longer needed in the [network auditor whitelist](https://github.com/brave/brave-browser/blob/629775164b8c596263fb7f5d2f8ee156dbb2cdc8/lib/whitelistedUrlPrefixes.js#L26-L28):

```
  'https://updates.bravesoftware.com/', // remove this once updates are moved to the prod environment
  'https://ads-serve.bravesoftware.com/', // remove this once ads catalog moves to using prod
  'https://pdfjs.robwu.nl/logpdfjs', // allowed because it gets canceled in tracking protection
```

### PR DESCRIPTION
**Pull request by fmarier**
Fixes #7789.

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection

## Test Plan:

`npm run network-audit`

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions.

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: lib/whitelistedUrlPrefixes.js [hunk 1]
```diff
@@ -23,9 +23,7 @@ module.exports = [
   'https://balance-staging.mercury.basicattentiontoken.org/',
   'https://publishers.basicattentiontoken.org/',
   'https://publishers-staging.basicattentiontoken.org/',
-  'https://updates.bravesoftware.com/', // remove this once updates are moved to the prod environment
-  'https://ads-serve.bravesoftware.com/', // remove this once ads catalog moves to using prod
```
Comment (diracdeltas) [inline lib/whitelistedUrlPrefixes.js]: what's the prod url for this? just curious. @amirsaber

Comment (fmarier) [reply lib/whitelistedUrlPrefixes.js]: `https://ads-serve.brave.com/v3/catalog` is the prod URL according to https://bravesoftware.slack.com/archives/CJR5902AW/p1579288365002100

Comment (diracdeltas) [reply lib/whitelistedUrlPrefixes.js]: should that be added to this list then?

Comment (diracdeltas) [reply lib/whitelistedUrlPrefixes.js]: nvm, just saw your comment about how the tests pass regardless. i wonder why it's not failing then? are ads not downloaded upon first browser startup?

Comment (diracdeltas) [reply lib/whitelistedUrlPrefixes.js]: maybe the issue is that rewards needs to be enabled before the catalog is downloaded, and this test doesn't automatically go through the onboarding process

Comment (fmarier) [reply lib/whitelistedUrlPrefixes.js]: > maybe the issue is that rewards needs to be enabled before the catalog is downloaded, and this test doesn't automatically go through the onboarding process

That's my guess as well.

**Comment by bsclifton** [comment]
Quick question about this process (examining whitelist + looking to remove unneeded / stale entries, etc)

Should it become a regular process? ex: once or twice a quarter, should this and other items (for example extension whitelist) be re-evaluated?

**Comment by fmarier** [comment]
> Should it become a regular process? ex: once or twice a quarter, should this and other items (for example extension whitelist) be re-evaluated? Either manually or via automation. We could keep a more detailed issue for each exception and check against that in an audit

I think that's a very good idea. These lists have a tendency to grow without bounds if left unchecked.

Perhaps the security team could own the quarterly audit of the network audit list.

**Pull request merged**